### PR TITLE
Bittrex Exchange

### PR DESCRIPTION
**Pull request by station384**
Correction for #1211, #1170, #1164 
Updated bittrex api from node.bittrex.api -> node-bittrex-api

node.bittrex.api is no longer maintained

Added some error handling so exchange is now passing back useful errors
to the engine.
i.e.  it will no longer keep retrying a buy sell order if there isn’t
enough funds.

socket errors no longer crash the system but passed to the retry
handler.

**npm must be updated.
npm install node-bittrex-api**

After these corrections it eliminated the errors cited in #1211. 

Unintentional corrections: (I like these :)
Errors now passed back to engine.js for proper handling
Orders will now readjust price ( old versions was silently failing )

### CODE REVIEW

File: extensions/exchanges/bittrex/exchange.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
-var bittrex_authed = require('node.bittrex.api'),
-    bittrex_public = require('node.bittrex.api'),
-    path = require('path'),
-    moment = require('moment'),
-    n = require('numbro'),
-    colors = require('colors')
+var bittrex_authed = require('node-bittrex-api'),
+  bittrex_public = require('node-bittrex-api'),
+  moment = require('moment'),
+  n = require('numbro')
```
Comment (DeviaVir) [inline extensions/exchanges/bittrex/exchange.js]: nit: indenting?

File: extensions/exchanges/bittrex/exchange.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
-var bittrex_authed = require('node.bittrex.api'),
-    bittrex_public = require('node.bittrex.api'),
-    path = require('path'),
-    moment = require('moment'),
-    n = require('numbro'),
-    colors = require('colors')
+var bittrex_authed = require('node-bittrex-api'),
```
Comment (DeviaVir) [inline extensions/exchanges/bittrex/exchange.js]: Should `historyScan` still be `forward`?
https://github.com/DeviaVir/zenbot/pull/1215/files#diff-b3b56e634e6ef011a9923a3c54c40108L51

File: extensions/exchanges/bittrex/exchange.js [hunk 3]
```diff
@@ -200,26 +233,50 @@ module.exports = function container(get, set, clear) {
...
-          return cb(null, [])
+          return cb(null, {})
         }
-
+      
+      
+ 
         if(!data.success) {
-          if (data.message && data.message.match(recoverableErrors)) {
+          if ( data.message && data.message.match(recoverableErrors)) {
```
Comment (DeviaVir) [inline extensions/exchanges/bittrex/exchange.js]: nit: unnecessary whitespace

**Review by DeviaVir — APPROVED** [review]

**Comment by station384** [comment]
will correct in a few.

File: extensions/exchanges/bittrex/exchange.js (hunk 2 above)
Comment (station384) [reply extensions/exchanges/bittrex/exchange.js]: Unsure.   The market history  on bittrex is very limited.   It will pull only the last couple of trades.  there is a workaround that someone came up with using the v2 api that seems to get around this limitation,  I'll look at implementing it.  hate having to leave a marking noop trading for days to get a decent history on.

File: extensions/exchanges/bittrex/exchange.js (hunk 1 above)
Comment (station384) [reply extensions/exchanges/bittrex/exchange.js]: It is formatted according the the setting in .eslintrc.
I'll manually format it.

File: extensions/exchanges/bittrex/exchange.js (hunk 2 above)
Comment (station384) [reply extensions/exchanges/bittrex/exchange.js]: I just double checked this   moving to forward causes 
on markets I already have data on
error: getTrades() returned duplicate results
{ product_id: 'XRP-USDT', to: 1516785694047 }
{ product_id: 'XRP-USDT', to: 1516785694047 }
it returns nothing on markets I don't have data on.

So for now, I think it should remain as backward until I can looking into the history calls in bittrex.

**Comment by station384** [comment]
@DeviaVir, it looks like I got all the corrections for formatting.   

The history part is just broken due to the api, and the way bittrex returns market data. 
I'll look at that later today and see if I can implement the a possible fix mentioned in #499.

I would also like to convert this to using websockets but that is for a later date.

**Comment by DeviaVir** [comment]
SGTM I will merge this now so we can look at the other additions fresh.
